src/main/java/com/mojang/datafixers/DataFixerBuilder.java

```diff
@@ -62,16 +63,36 @@ public DataFixer build(final Executor executor) {
         final DataFixerUpper fixerUpper = new DataFixerUpper(new Int2ObjectAVLTreeMap<>(schemas), new ArrayList<>(globalList), new IntAVLTreeSet(fixerVersions));
 
         executor.execute(() -> {
+            List<Runnable> allTasks = new ArrayList<>();
             final IntBidirectionalIterator iterator = fixerUpper.fixerVersions().iterator();
             while (iterator.hasNext()) {
                 final int versionKey = iterator.nextInt();
                 final Schema schema = schemas.get(versionKey);
                 for (final String typeName: schema.types()) {
-                    final Type<?> dataType = schema.getType(() -> typeName);
-                    final TypeRewriteRule rule = fixerUpper.getRule(DataFixUtils.getVersion(versionKey), dataVersion);
-                    dataType.rewrite(rule, DataFixerUpper.OPTIMIZATION_RULE);
+                    allTasks.add(() -> {
+                        final Type<?> dataType = schema.getType(() -> typeName);
+                        final TypeRewriteRule rule = fixerUpper.getRule(DataFixUtils.getVersion(versionKey), dataVersion);
+                        dataType.rewrite(rule, DataFixerUpper.OPTIMIZATION_RULE);
+                    });
                 }
             }
+
+            // Divide up into sets of tasks by number of CPU cores
+            // Some tasks are faster than others, randomize it to try to divide it more
+            Collections.shuffle(allTasks);
+            List<List<Runnable>> queueList = new ArrayList<>();
+            List<Runnable> current = new ArrayList<>();
+            queueList.add(current);
+            int maxTasks = (int) Math.max(1, Math.floor(allTasks.size() / (float)Math.min(6, Runtime.getRuntime().availableProcessors()-2)));
```

Comment:
I decided to do -2 on core count to not hog every thread, leaving 1 for game and 1 extra as a max, and an overall cap of 6 for systems with many cores overall.